core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamUnitTest.java

```diff
@@ -0,0 +1,86 @@
+package com.baeldung.inputstream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+public class InputStreamUnitTest {
+    @Test
+    public void givenAStringWrittenToFile_whenReadWithFileInputStream_thenItShouldExitsInTheInputStream(@TempDir Path tempDir) throws IOException {
+        Path sampleOut = tempDir.resolve("sample-out.txt");
+        List<String> lines = Arrays.asList("Hello. This is just a test. Good bye.");
+        Files.write(sampleOut, lines);
+        File sampleOutFile = sampleOut.toFile();
+        try (InputStream inputStream = new FileInputStream(sampleOutFile)) {
+            assertThat(readString(inputStream)).contains(lines.get(0));
+        }
+    }
+
+    @Test
+    public void givenAStringWrittenToFile_whenReadWithFileInputStreamWithStringConstructor_thenItShouldExitsInTheInputStream(@TempDir Path tempDir) throws IOException {
+        Path sampleOut = tempDir.resolve("sample-out.txt");
+        String expectedText = "Hello. Hi.";
+        List<String> lines = Arrays.asList(expectedText);
+        Files.write(sampleOut, lines);
+        String fileAbsolutePath = sampleOut.toFile()
+            .getAbsolutePath();
+        try (FileInputStream fis = new FileInputStream(fileAbsolutePath)) {
+            int availableBytes = fis.available();
+            assertThat(availableBytes > 0);
```

Comment:
This is not a complete assertion, you need something like:
```suggestion
            assertThat(availableBytes).isGreaterThan(0);
```